ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParsePayload.java

```diff
@@ -101,17 +101,16 @@ private ParsePayload(ValueProvider<String> schemasLocation) {
           // been applied, we strip out any existing metadata fields and put them into attributes.
           AddMetadata.stripPayloadMetadataToAttributes(attributes, json);
 
-          // Prevent message that need to be scrubbed from going to success.
+          // Check the contents of the message, potentially throwing an exception that causes the
+          // message to be dropped or routed to error output; may also also alter the payload to
+          // redact sensitive fields.
           try {
             MessageScrubber.scrub(attributes, json);
           } catch (MessageShouldBeDroppedException e) {
             // This message should go to no output, so we return an empty list immediately.
             return Collections.emptyList();
           }
 
-          // Potentially mutates the value of json to redact specific fields.
-          MessageScrubber.redact(attributes, json);
```

Comment:
With the refactoring of MessageScrubber and grouping of like actions together, it seems less necessary to have `redact` be a separate method from `scrub`.